### ISSUE TITLE
Notated moves allows every piece to be promoted following a promotable pawn

### ISSUE THREAD

**Issue by CFT-Chris** [brozeph/node-chess#17]
Steps to reproduce:

1. Move a white pawn up to the seventh rank.
2. Look at the notated moves afterward -- every white piece that is notated after that pawn (all minor/major pieces, not just pawns) have promotions attached to their notated moves.

Suggested fix below:

node-chess/lib/algebraicGameClient.js (lines 79 to 86)

```javascript
// iterate through each starting squares valid moves
	for (i = 0; i < validMoves.length; i++) {
		p = validMoves[i].src.piece;

		// iterate each potential move and build prefix and suffix for notation
		for (n = 0; n < validMoves[i].squares.length; n++) {
			prefix = '';

            isPromotion = false; // <---- need to reset isPromotion
```

**Comment by brozeph** [comment]
@CFT-Chris, I'm having a tough time reproducing, but I'm continuing to test. Here's my first attempt to reproduce, but the output is as expected:

```javascript
var 
  gc = algebraicGameClient.create(),
  status;

  // position the board for a promotion next move
  gc.game.board.getSquare('c7').piece = null;
  gc.game.board.getSquare('c8').piece = null;
  gc.game.board.getSquare('c2').piece = null;
  gc.game.board.getSquare('c7').piece = piece.createPawn(piece.SideType.White);
  gc.game.board.getSquare('c7').piece.moveCount = 1;

  // force recalculation of board position
  status = gc.getStatus(true);

  console.log(Object.keys(status.notatedMoves));
```

Produces the following output:

```javascript
[ 'Na3',
  'Nc3',
  'Qc2',
  'Qb3',
  'Qa4',
  'Nf3',
  'Nh3',
  'a3',
  'a4',
  'b3',
  'b4',
  'd3',
  'd4',
  'e3',
  'e4',
  'f3',
  'f4',
  'g3',
  'g4',
  'h3',
  'h4',
  'c8R',
  'c8N',
  'c8B',
  'c8Q',
  'cxb8R',
  'cxb8N',
  'cxb8B',
  'cxb8Q',
  'cxd8R',
  'cxd8N',
  'cxd8B',
  'cxd8Q' ]
```

**Comment by CFT-Chris** [comment]
@brozeph, if I add a minor/major piece that has its moves calculated after the promotable pawn, then I can see the issue reproduce.  In your test case I added a bishop to h7:

```javascript
// position the board for a promotion next move
  gc.game.board.getSquare('c7').piece = null;
  gc.game.board.getSquare('c8').piece = null;
  gc.game.board.getSquare('c2').piece = null;
  gc.game.board.getSquare('c7').piece = piece.createPawn(piece.SideType.White);
  gc.game.board.getSquare('c7').piece.moveCount = 1;
  gc.game.board.getSquare('h7').piece = null;
  gc.game.board.getSquare('h7').piece = piece.createBishop(piece.SideType.White);
  gc.game.board.getSquare('h7').piece.moveCount = 1;
  
  // force recalculation of board position
  status = gc.getStatus(true);

  console.log(Object.keys(status.notatedMoves));
```

and it produced the following output:
```
[ 'Na3',
  'Nc3',
  'Qc2',
  'Qb3',
  'Qa4',
  'Nf3',
  'Nh3',
  'a3',
  'a4',
  'b3',
  'b4',
  'd3',
  'd4',
  'e3',
  'e4',
  'f3',
  'f4',
  'g3',
  'g4',
  'h3',
  'h4',
  'c8R',
  'c8N',
  'c8B',
  'c8Q',
  'cxb8R',
  'cxb8N',
  'cxb8B',
  'cxb8Q',
  'cxd8R',
  'cxd8N',
  'cxd8B',
  'cxd8Q',
  'Bxg8R',
  'Bxg8N',
  'Bxg8B',
  'Bxg8Q',
  'Bg6R',
  'Bg6N',
  'Bg6B',
  'Bg6Q',
  'Bf5R',
  'Bf5N',
  'Bf5B',
  'Bf5Q',
  'Be4R',
  'Be4N',
  'Be4B',
  'Be4Q',
  'Bd3R',
  'Bd3N',
  'Bd3B',
  'Bd3Q',
  'Bc2R',
  'Bc2N',
  'Bc2B',
  'Bc2Q' ]
```

The QRNB promotion characters are appended to the notated moves of the bishop.  Thanks for putting this module up and taking the time to look at this.

**Comment by brozeph** [comment]
Hi @CFT-Chris, thank you for the reproduction steps... I'm still having a challenge reproducing though. When I run the above code with `v0.2.6` of the module I see the following output:

```javascript
[ 'Na3',
  'Nc3',
  'Nf3',
  'Nh3',
  'a3',
  'a4',
  'b3',
  'b4',
  'c3',
  'c4',
  'd3',
  'd4',
  'e3',
  'e4',
  'f3',
  'f4',
  'g3',
  'g4',
  'h3',
  'h4' ]
```

**Comment by brozeph** [comment]
Update: Got it to repro! Working on fix now!

**Comment by brozeph** [comment]
@CFT-Chris - great bug report and repro steps - thank you! This is now resolved in `v0.2.7` and published to NPM.
